source/clear-linux/guides/maintenance/swupd-mixer-integration.rst

```diff
@@ -0,0 +1,155 @@
+default build of |CLOSIA| provides options to install bundles for various
+server capabilities. If you want to supplement the default OS with
+functionality from other OS distributions, you can manually compose an OS
+using :ref:`the mixer tool <mixer>`.
+
+Software updater release 19900 integrates the mixer tool with the
+updater (:command:`swupd`) and provides a simpler way to mix your OS.
+
+Process overview
+****************
+
+To add a package that does not exist in the current |CL| bundle, perform the following:
+
+#. Create custom packages and bundles using the :file:`swupd-add-pkg` script.
+
+#. Add your desired content with the commands:
+
+   .. code-block:: bash
+
+      sudo swupd update
+      sudo swupd bundle add <new_bundle>
+
+The upstream content is pulled from the normal URL and local content is added
+from a local update store.
+
+.. note::
+
+   The integrated mixer tool adds non-unique packages and replaces upstream
+   content with a local version. However, the updater exits if it finds
+   conflicts before creating the local content. You can override this behavior
+   by passing the :command:`--allow-mix-collisions` option.
+
+
+Prerequisites
+*************
+
+To use the integrated swupd mixer, you need a recent image of
+|CLOSIA| with the mixer bundle installed. If you do not have it already,
+add it with the :command:`swupd bundle-add` command as shown below:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add mixer
+
+
+Using the enhanced swupd mixer
+******************************
+
+Perform the following steps:
+
+#. Create a folder :file:`/usr/share/mix/rpms` for the local update content
+   and copy the desired RPMs into the folder:
+
+   .. note::
+
+     The RPMs must be formatted and built for |CL|.
+
+   .. code-block:: bash
+
+      sudo cp <RPM1 RPMn> /usr/share/mix/rpms/
+
+#. Add RPMs and create content.
+
+   .. code-block:: bash
+
+      sudo swupd-add-pkg <RPM1 RPMn> <bundle_name>
+
+   The :file:`swupd-add-pkg` script creates the :command:`<bundle_name>`
+   bundle if it does not exist, adds the RPMs to it, and generates the
+   required mixer content and manifests.
+
+#. Use the new :command:`--migrate` flag to let :command:`swupd` know that
+   local content exists.
+
+   .. code-block:: bash
+
+      sudo swupd update --migrate
+
+   When complete, your system is in the mix-ready state using the new
+   Manifest.MoM.
+
+#. Add your bundle to your system:
+
+   .. code-block:: bash
+
+      sudo swupd bundle-add <bundle_name>
+
+
+Reverting to the upstream version
+*********************************
+
+You can return your OS to an official upstream version with no user-added
+content using the following command:
+
+   .. code-block:: bash
+
+      sudo swupd verify --fix --picky --force -m <upstreamversion> -C
+      /usr/share/clear/update-ca/Swupd_Root.pem
+
+   .. note::
+
+      The options used in the example delete the :file:`/usr/share/mix` folder.
+
+
+swupd operational details
```

Comment:
Text from line 111-155 is very detailed and may not be relevant for all users, so I moved it to the end of this topic, OK?

Reply:
I think this is acceptable, good idea!